Source/mpqapi.cpp

```diff
@@ -578,30 +578,28 @@ BOOLEAN __fastcall mpqapi_parse_archive_header(_FILEHEADER *pHdr, int *pdwNextFi
 
 void __fastcall mpqapi_close_archive(const char *pszArchive, BOOL bFree, int dwChar) // CloseMPQ
 {
-	const char *v3;  // esi
-	_BLOCKENTRY *v4; // ecx
-	_HASHENTRY *v5;  // ecx
+	_BLOCKENTRY *blockEntry;
+	_HASHENTRY *hashEntry;
 
-	v3 = pszArchive;
 	if (bFree) {
-		v4 = sgpBlockTbl;
+		blockEntry = sgpBlockTbl;
 		sgpBlockTbl = 0;

```

Comment:
This is clearing the pointer after freeing it, so it should be styled as NULL:
```suggestion
		sgpBlockTbl = NULL;
```

Reply:
Additionally, you can remove those three lines and use `MemFreeDbg` macro =)